async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -499,6 +541,21 @@ impl futures::Stream for Stream {
                 self.request_tx.send(()).unwrap();
                 self.pending_request = true;
             }
+            match self.hearbeats_missing.try_recv() {
+                Ok(_) => {
+                    return Poll::Ready(Some(Err(Box::new(std::io::Error::new(
+                        std::io::ErrorKind::TimedOut,
+                        "did not receive idle hearbeat in time",
+                    )))))
+                }
+                Err(TryRecvError::Empty) => (),
```

Comment:
Note/comment why this is ignored.

```suggestion
                Err(TryRecvError::Empty) => (),
```